### PR TITLE
SplitWhen/After and GroupBy implementation

### PR DESCRIPTION
**Pull request by cconstantin**
GroupBy has some issues, will investigate tomorrow.

### CODE REVIEW

File: src/core/Akka.Streams/Dsl/Source.cs [hunk 1]
```diff
@@ -134,6 +134,14 @@ public IRunnableGraph<TMat> To<TMat2>(IGraph<SinkShape<TOut>, TMat2> sink)
             return new Source<TOut, TMat2>(Module.TransformMaterializedValue(mapFunc));
         }
 
+        internal Source<TOut2, TMat> DeprecatedAndThen<TOut2>(StageModule op)
```
Comment (Horusiath) [inline src/core/Akka.Streams/Dsl/Source.cs]: Do we really need to port deprecated API?

**Comment by Horusiath** [comment]
Beside one question attached, the rest is good to go.

File: src/core/Akka.Streams/Dsl/Source.cs (hunk 1 above)
Comment (cconstantin) [inline src/core/Akka.Streams/Dsl/Source.cs]: It's still there in the latest version, and I'm not sure which direction they're taking.

**Comment by cconstantin** [comment]
@Horusiath let's hold off on merging this, while investigating the remaining GroupBy issues I found that there may be an easier way to implement GroupBy. I'll post a comment when ready.

**Comment by cconstantin** [comment]
GroupBy fixed and all tests pass, ready to merge.

**Pull request merged**